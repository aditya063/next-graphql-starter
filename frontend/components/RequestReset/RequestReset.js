import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';

import DisplayError from '../DisplayError/DisplayError';
import { REQUEST_RESET_MUTATION } from '../../graphql/queries';
import * as sc from './RequestReset.style';

class RequestReset extends React.PureComponent {
  componentDidMount() {
    // Disable submit button at the beginning
    this.props.form.validateFields();
  }

  hasErrors = fieldsError =>
    Object.keys(fieldsError).some(field => fieldsError[field]);

  handleSubmitForm = (e, requestReset) => {
    e.preventDefault();

    this.props.form.validateFields((err, values) => {
      requestReset({ variables: values });

      this.props.form.resetFields();
    });
  };

  handleUpdate = cache => cache.writeData({ data: { isLoggedIn: true } });

  handleError = error => error;

  render() {
    const {
      getFieldDecorator,
      getFieldsError,
      getFieldError,
      isFieldTouched
    } = this.props.form;

    // Only show error after a field is touched.
    const emailError = isFieldTouched('email') && getFieldError('email');

    return (
      <Mutation
        mutation={REQUEST_RESET_MUTATION}
        variables={this.state}
        onError={this.handleError}
      >
        {(requestReset, { loading, error, called }) => (
          <sc.RequestReset
            onSubmit={e => this.handleSubmitForm(e, requestReset)}
          >
            <sc.FormItem
              validateStatus={emailError ? 'error' : ''}
              help={emailError || ''}
              hasFeedback
            >
              {getFieldDecorator('email', {
                rules: [
                  { required: true, message: 'Please enter your email' },
                  { type: 'email', message: 'Not a valid email address' }
                ]
              })(
                <sc.InputEmail
                  type="email"
                  placeholder="email"
                  onPressEnter={e => this.handleSubmitForm(e, signIn)}
                  prefix={(
                    <sc.InputIcon
                      type="user"
                      style={{ color: 'rgba(0,0,0,.25)' }}
                    />
)}
                />
              )}
            </sc.FormItem>

            {error && <DisplayError error={error} />}

            {!error && !loading && called && (
              <p>Check your email for a reset link.</p>
            )}

            <sc.FormItem>
              <sc.SubmitBtn
                loading={loading}
                type="primary"
                htmlType="submit"
                disabled={this.hasErrors(getFieldsError())}
              >
                Submit
              </sc.SubmitBtn>
            </sc.FormItem>
          </sc.RequestReset>
        )}
      </Mutation>
    );
  }
}

RequestReset.propTypes = {
  form: PropTypes.shape({
    validateFields: PropTypes.func.isRequired,
    getFieldDecorator: PropTypes.func.isRequired,
    getFieldsError: PropTypes.func.isRequired,
    getFieldError: PropTypes.func.isRequired,
    isFieldTouched: PropTypes.func.isRequired,
    resetFields: PropTypes.func.isRequired
  }).isRequired
};

export default sc.RequestReset.create({ name: 'RequestReset' })(RequestReset);
