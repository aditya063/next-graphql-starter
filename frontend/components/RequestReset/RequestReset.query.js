import gql from 'graphql-tag';

export const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    requestReset(email: $email)
  }
`;

export const MOCK_REQUEST_RESET_MUTATION = {
  request: {
    query: REQUEST_RESET_MUTATION,
    variables: { email: 'mock@email.com' }
  },
  result: {
    data: {
      requestReset: true
    }
  }
};
