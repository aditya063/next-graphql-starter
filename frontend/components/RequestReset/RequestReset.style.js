import styled from 'styled-components';
import { Form, Input, Button, Icon } from 'antd';

export const RequestReset = styled(Form)``;

export const FormItem = styled(Form.Item)``;

export const InputIcon = styled(Icon)`
  color: rgba(0, 0, 0, 0.5);
`;

export const InputEmail = styled(Input)``;

export const SubmitBtn = styled(Button)`
  float: right;
`;
