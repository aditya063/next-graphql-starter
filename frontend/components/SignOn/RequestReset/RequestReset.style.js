import styled from 'styled-components';

export const div = styled.div`
  fieldset {
    display: flex;
    * {
      margin: 4px;
    }

    button {
      align-self: center;
    }
  }
`;
