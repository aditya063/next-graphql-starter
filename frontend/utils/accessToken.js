import fetch from 'isomorphic-unfetch';
// import cookie from 'cookie';
// import jwt from 'jsonwebtoken';

let accessToken = '';

export const setAccessToken = token => (accessToken = token);

export const getAccessToken = () => accessToken;

export const clearAccessToken = () => (accessToken = '');

export const fetchAccessToken = async refreshToken => {
  const url =
    process.env.NODE_ENV === 'development'
      ? process.env.DEV_REFRESH_URL
      : process.env.PROD_REFRESH_URL;

  const options = {
    method: 'GET',
    credentials: 'include',
    headers: { cookie: `rt=${refreshToken}` }
  };

  try {
    const response = await fetch(url, options);

    const data = await response.json();

    return data.accessToken;
  } catch (error) {
    if (process.env.NODE_ENV === 'development')
      console.error('fetchAccessToken error: ', error);
  }
};

// export const verifyAccessToken = accessToken => {
//   try {
//     jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET);
//   } catch (error) {
//     if (process.env.NODE_ENV === 'development')
//       console.error('fetchAccessToken error: ', error);
//   }
// };

// const checkAccessToken = async cookies => {
//   const accessToken = getAccessToken();

//   if (!accessToken) return true;

//   const parsedCookies = cookie.parse(cookies);

//   if (!parsedCookies.rt) return '';

//   // const response = await fetchAccessToken(parsedCookies.rt);

//   // const data = await response.json();

//   // setAccessToken(data.accessToken);
//   // setAccessTokenExpiry(data.exp);

//   // return data.accessToken;
// };
