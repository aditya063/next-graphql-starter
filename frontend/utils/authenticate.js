import jwt from 'jsonwebtoken';

const authenticate = cookie => {
  const secret = process.env.JWT_SECRET;
  const token = cookie.slice(6);

  try {
    jwt.verify(token, secret);

    return true;
  } catch {
    return false;
  }
};

const setCache = (cache, isAuthenticated) => {
  cache.writeData({
    data: {
      isLoggedIn: isAuthenticated
    }
  });
};

export default (cache, cookie) => {
  const isAuthenticated = cookie ? authenticate(cookie) : false;

  setCache(cache, isAuthenticated);
};
