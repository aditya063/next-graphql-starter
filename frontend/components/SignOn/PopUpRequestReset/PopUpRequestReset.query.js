import { REQUEST_RESET_MUTATION } from '../../RequestReset/RequestReset.query';

export { REQUEST_RESET_MUTATION };
