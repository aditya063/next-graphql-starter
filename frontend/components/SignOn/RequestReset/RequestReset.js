import { Mutation } from 'react-apollo';
import Error from '../../Error/Error';
import * as query from './RequestReset.query';

class RequestReset extends React.PureComponent {
  state = {
    email: ''
  };

  onChangeEmail = e => this.setState({ [e.target.name]: e.target.value });

  render() {
    return (
      <Mutation mutation={query.REQUEST_RESET_MUTATION} variables={this.state}>
        {(requestReset, { error, loading, called }) => (
          <form
            method="post"
            data-test="form"
            onSubmit={async e => {
              e.preventDefault();
              await requestReset();
              this.setState({ email: '' });
            }}
          >
            <fieldset disabled={loading} aria-busy={loading}>
              <h2>Request a password reset</h2>
              <Error error={error} />
              {!error && !loading && called && (
                <p>Check your email for a reset link.</p>
              )}

              <label htmlFor="email">
                Email
                <input
                  type="email"
                  name="email"
                  placeholder="email"
                  value={this.state.email}
                  onChange={this.onChangeEmail}
                />
              </label>

              <button type="submit">Request Reset</button>
            </fieldset>
          </form>
        )}
      </Mutation>
    );
  }
}

export default RequestReset;
