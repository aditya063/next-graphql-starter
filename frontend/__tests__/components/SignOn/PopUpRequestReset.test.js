import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { ThemeProvider } from 'styled-components';
import { load } from '../../../utils/load';
// eslint-disable-next-line max-len
import PopUpRequestReset from '../../../components/SignOn/PopUpRequestReset/PopUpRequestReset';
// eslint-disable-next-line max-len
import { MOCK_REQUEST_RESET_MUTATION } from '../../../components/RequestReset/RequestReset.query';
import theme from '../../../styles/theme.style';

describe('ResetPassword', () => {
  let mockProps;
  let mockQueries;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [MOCK_REQUEST_RESET_MUTATION];
    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <ThemeProvider theme={theme}>
          <PopUpRequestReset {...mockProps} />
        </ThemeProvider>
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot', () => {
    const wrapSnap = wrapper.find('formstyle__form');

    // console.log(wrapper.debug());

    expect(wrapSnap).toMatchSnapshot();
  });

  it('matches snapshot - loaded', async () => {
    const wrapSnap = wrapper.find('formstyle__form');

    await load(wrapper);

    // console.log(wrapper.debug());

    expect(wrapSnap).toMatchSnapshot();
  });
});
