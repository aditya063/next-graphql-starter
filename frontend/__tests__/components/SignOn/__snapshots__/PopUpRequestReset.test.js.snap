// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword matches snapshot - loaded 1`] = `
.c0 > fieldset {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border: 1px solid #393939;
  padding: 4px;
  border-radius: 2px;
  margin: 5px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 1px 1px 0 #fff;
  font-size: 1rem;
  font-weight: 600;
  padding: 0.3rem 1rem;
  border: 2px solid;
  border-radius: 3px;
  cursor: pointer;
  -webkit-transition: background-color 0.3s ease-out;
  transition: background-color 0.3s ease-out;
  background: none;
  white-space: nowrap;
  margin-top: 10px;
  margin-left: auto;
}

.c2:hover {
  background-color: #eee;
}

.c2:active {
  box-shadow: 0 2px 6px rgba(0,0,0,0.25) inset;
}

<formstyle__form
  onSubmit={[Function]}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "formstyle__form-q97ky5-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "> fieldset{display:flex;}",
          ],
        },
        "displayName": "formstyle__form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "formstyle__form-q97ky5-0",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onSubmit={[Function]}
  >
    <form
      className="c0"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Request a password reset
        </h2>
        <label
          htmlFor="email"
        >
          Email
          <PopUpRequestResetstyle__inputText
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "PopUpRequestResetstyle__inputText-rbxiap-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "PopUpRequestResetstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "PopUpRequestResetstyle__inputText-rbxiap-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            >
              <input
                className="c1"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
              />
            </StyledComponent>
          </PopUpRequestResetstyle__inputText>
        </label>
        <PopUpRequestResetstyle__inputBtn
          disabled={true}
          type="submit"
          value="Request Reset"
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "PopUpRequestResetstyle__inputBtn-rbxiap-1",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "text-decoration:none;text-shadow:1px 1px 0 #fff;font-size:1rem;font-weight:600;padding:0.3rem 1rem;border:2px solid;border-radius:3px;cursor:pointer;transition:background-color 0.3s ease-out;background:none;white-space:nowrap;&:hover{background-color:#eee;}&:active{box-shadow:0 2px 6px rgba(0,0,0,0.25) inset;}",
                    "color:",
                    [Function],
                    ";",
                    "margin-top:10px;margin-left:auto;",
                  ],
                },
                "displayName": "PopUpRequestResetstyle__inputBtn",
                "foldedComponentIds": Array [
                  "buttonstyle__inputBtn-sc-5slzqu-3",
                  "buttonstyle__submitInputBtn-sc-5slzqu-5",
                ],
                "render": [Function],
                "styledComponentId": "PopUpRequestResetstyle__inputBtn-rbxiap-1",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            type="submit"
            value="Request Reset"
          >
            <input
              className="c2"
              disabled={true}
              type="submit"
              value="Request Reset"
            />
          </StyledComponent>
        </PopUpRequestResetstyle__inputBtn>
      </fieldset>
    </form>
  </StyledComponent>
</formstyle__form>
`;

exports[`ResetPassword matches snapshot 1`] = `
.c0 > fieldset {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border: 1px solid #393939;
  padding: 4px;
  border-radius: 2px;
  margin: 5px;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-shadow: 1px 1px 0 #fff;
  font-size: 1rem;
  font-weight: 600;
  padding: 0.3rem 1rem;
  border: 2px solid;
  border-radius: 3px;
  cursor: pointer;
  -webkit-transition: background-color 0.3s ease-out;
  transition: background-color 0.3s ease-out;
  background: none;
  white-space: nowrap;
  margin-top: 10px;
  margin-left: auto;
}

.c2:hover {
  background-color: #eee;
}

.c2:active {
  box-shadow: 0 2px 6px rgba(0,0,0,0.25) inset;
}

<formstyle__form
  onSubmit={[Function]}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "formstyle__form-q97ky5-0",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "> fieldset{display:flex;}",
          ],
        },
        "displayName": "formstyle__form",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "formstyle__form-q97ky5-0",
        "target": "form",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    onSubmit={[Function]}
  >
    <form
      className="c0"
      onSubmit={[Function]}
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Request a password reset
        </h2>
        <label
          htmlFor="email"
        >
          Email
          <PopUpRequestResetstyle__inputText
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "PopUpRequestResetstyle__inputText-rbxiap-0",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "border:1px solid ",
                      [Function],
                      ";padding:4px;border-radius:2px;",
                      "margin:5px;",
                    ],
                  },
                  "displayName": "PopUpRequestResetstyle__inputText",
                  "foldedComponentIds": Array [
                    "inputstyle__input-bgiqmz-0",
                  ],
                  "render": [Function],
                  "styledComponentId": "PopUpRequestResetstyle__inputText-rbxiap-0",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
            >
              <input
                className="c1"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
              />
            </StyledComponent>
          </PopUpRequestResetstyle__inputText>
        </label>
        <PopUpRequestResetstyle__inputBtn
          disabled={true}
          type="submit"
          value="Request Reset"
        >
          <StyledComponent
            disabled={true}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "PopUpRequestResetstyle__inputBtn-rbxiap-1",
                  "isStatic": false,
                  "lastClassName": "c2",
                  "rules": Array [
                    "text-decoration:none;text-shadow:1px 1px 0 #fff;font-size:1rem;font-weight:600;padding:0.3rem 1rem;border:2px solid;border-radius:3px;cursor:pointer;transition:background-color 0.3s ease-out;background:none;white-space:nowrap;&:hover{background-color:#eee;}&:active{box-shadow:0 2px 6px rgba(0,0,0,0.25) inset;}",
                    "color:",
                    [Function],
                    ";",
                    "margin-top:10px;margin-left:auto;",
                  ],
                },
                "displayName": "PopUpRequestResetstyle__inputBtn",
                "foldedComponentIds": Array [
                  "buttonstyle__inputBtn-sc-5slzqu-3",
                  "buttonstyle__submitInputBtn-sc-5slzqu-5",
                ],
                "render": [Function],
                "styledComponentId": "PopUpRequestResetstyle__inputBtn-rbxiap-1",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            type="submit"
            value="Request Reset"
          >
            <input
              className="c2"
              disabled={true}
              type="submit"
              value="Request Reset"
            />
          </StyledComponent>
        </PopUpRequestResetstyle__inputBtn>
      </fieldset>
    </form>
  </StyledComponent>
</formstyle__form>
`;
